### DISCUSSION TITLE
Case insensitive queries

### DISCUSSION THREAD

**Discussion by patrickxchong** [deta/cloud-docs#410] [Help]
Hello, based on the [documentation](https://docs.deta.sh/docs/base/queries#contains) it seems like Deta base does not support case insensitive queries yet, is that true? If yes, is the feature in the pipeline? Need it to power a case insensitive search feature.

**Comment by jnsougata** [comment]
Can you please elaborate wdym by the `case sensitive` queries?

**Reply by patrickxchong** [reply]
I'm looking for a query that can match 
1. `andy`
2. `Andy`
3. `ANDY`
4. `aNDy` 
etc

when I used `andy` as the search term. below is an example using the `contains` query but that only matches the first example.
```
SongDB.fetch({"name?contains":"andy"})
```

A regex query would solve the problem as well but I don't see an option to use regex with Deta Base.

**Comment by aavshr** [comment]
hi @patrickxchong thanks for the request, `contains` is case sensitive at the moment.

**Comment by abdelhai** [comment]
idea: store data in lower case, maybe in a separate field
